### DISCUSSION TITLE
Silent install of Rancher v1.21 on MacOS

### DISCUSSION THREAD

**Discussion by LtCmdrScott** [rancher-sandbox/rancher-desktop#2090] [Q&A] [answered]
In our rather large corp environment, the MacOS end users are NOT allowed to have admin rights and we have to install all software via JAMF pro or WS1.  Is it possible to do a silent install of all the apps/files necessary, and the end user will NOT be asked for admin rights upon first launch?   All Mac users are on Big Sur or Monterey at this time.

**Comment by jandubois** [comment]
In the next release you will still be asked, but you can select an "Always run without administrative access" checkbox, and will not be asked again afterwards:

![Screen Shot 2022-04-26 at 5 12 07 PM](https://user-images.githubusercontent.com/78947/165413209-df6cce60-b3ee-4416-b565-85c1bcf84b17.png)

Of course if you run without admin rights then we can't use Apple's vmnet framework, so you will not get a proper IP address for the VM, but must use the forwarded ports for everything.

**Reply by LtCmdrScott** [reply]
that is good news...so will this new release allow me to do a completely silent install for MacOS then?  Again, the users have no admin rights, so after the install on their machine, it just "has to work".  Also...when is this release slated for download?

Thanks for giving me this feedback.  It helps knowing that a potential fix is in the works.

**Reply by LtCmdrScott** [reply]
ok...v1.30 is a huge improvement of 1.21 as far as silent install goes, but I do have  one more question...where are the settings file for that "ask for admin" (shown above) screen, so I can deliver that settings file to the user so they don't get prompt for it?  If end users see that, they will inadvertently try to allow "sudo" access and or course it won't work since they don't have admin rights and they will end up creating trouble tickets for our team to "fix" and in an organization of our size, we are trying to prevent users from generating trouble tickets

**Reply by jandubois** [reply]
They are stored in `~/Library/Preferences/rancher-desktop/settings.json` in the `kubernetes.suppressSudo` property. You will have to set it to `true`:

```console
$ jq .kubernetes.suppressSudo < ~/Library/Preferences/rancher-desktop/settings.json
false
```

Note that we are going to refactor the whole settings mechanism in the near future, so at least the structure of the file is going to change in one of the next releases (but not yet in 1.4.0).

**Reply by LtCmdrScott** [reply]
awesome!  thanks for so much for the fast reply and excellent support!

**Reply by SamStenton** [reply]
hey @LtCmdrScott did you do any more on this? Similar situation with users, but want to do the sudo actions for them on install so they don't have to port forward.

**Reply by LtCmdrScott** [reply]
I was able to do pretty much everything that I needed to with v1.3.  I am not getting the sudo access prompt after I do the install...it is my understand that the v1.4 that they are working on will provide even more non-admin possibility...

**Reply by SamStenton** [reply]
I assume that still requires users to port forward to access resources?

**Reply by LtCmdrScott** [reply]
yes, but that can be done without sudo in our case

**Reply by jandubois** [reply]
Just a quick note that v1.4.1 is the latest released version.

Note that the switch to a privileged helper process, so you can install with sudo, but not require sudo at runtime is still the plan, but not yet implemented (or even scheduled).

**Comment by jandubois** [comment, marked as answer]
Administrative Access is no longer the default, and is generally not needed, except for

* creating `/var/run/docker.sock` instead of using the `rancher-desktop` docker context
* using `socket_vmnet` to give the VM a bridged network interface on the local network

The admin prompt is never shown unless the user explicitly tries to enable administrative access.
